infrastructure/storageprovider/dropbox.go

```diff
@@ -89,11 +107,19 @@ func (d *dropbox) Upload(cred domain.StorageProviderCredential, file io.Reader,
 	}
 
 	// do the request
-	_, err = client.Do(req)
+	res, err := client.Do(req)
 	if err != nil {
 		return err
 	}
 
+	if res.StatusCode != http.StatusOK {
+		byteRes, _ := io.ReadAll(res.Body)
```

Comment:
are you sure there will be no error from io module for ReadAll response?
my concern is if there is an error may causing panic here, but if panic already handled it should be safe.

Reply:
I'm not so sure. Should I add an `if err != nil` block after ReadAll?

Reply:
yup

Reply:
Okay